app/controllers/spree/admin/invoices_controller.rb

```diff
@@ -19,8 +19,14 @@ def show
 
       def generate
         @order = Order.find_by(number: params[:order_id])
-        authorize! :invoice, @order
-        OrderInvoiceGenerator.new(@order).generate_or_update_latest_invoice
+        if @order.distributor.can_invoice?
+          authorize! :invoice, @order
+          OrderInvoiceGenerator.new(@order).generate_or_update_latest_invoice
+        else
+          flash[:error] = t(:must_have_valid_business_number,
+                            enterprise_name: @order.distributor.name)
+        end
```

Comment:
My first thought was why we don't include that check in the Abilities class to have one source of truth. But that wouldn't provide us with a good error message. I guess that's one downside of CanCan.